### PR TITLE
daemon returns errors when receiving unsupported prune filters

### PR DESCRIPTION
**Pull request by gdevillele**
**- What I did**

- container prune
- volume prune
- image prune
- network prune

...now return errors if an unsupported filter is received.

container prune only accepts the following filters:
- label
- until

volume prune only accepts the following filters:
- label

image prune only accepts the following filters:
- dangling
- label
- until

network prune only accepts the following filters:
- label
- until

**- How to verify it**

Example:

`docker container prune --filter <filter>=<key>=<value>`

if <filter> is not "label" or "until", you should get a daemon error.

**- Description for the changelog**

return errors to the client when receiving unsupported prune filters

**- A picture of a cute animal (not mandatory but encouraged)**

gp-7

cc @thaJeztah @vdemeester

### CODE REVIEW

File: daemon/prune.go [hunk 1]
```diff
@@ -25,6 +25,27 @@ var (
 	// ErrPruneRunning is returned when a prune request is received while
 	// one is in progress
 	ErrPruneRunning = fmt.Errorf("a prune operation is already running")
```
Comment (allencloud) [inline daemon/prune.go]: I am also wondering if we could disable the export of error `ErrPruneRunning `? 
Since it is only used here.
But maybe out of the scope of this PR. 🐱

File: daemon/prune.go [hunk 2]
```diff
@@ -25,6 +25,27 @@ var (
...
+
+	containersAcceptedFilters = map[string]bool{
+		"label":  true,
+		"label!": true,
```
Comment (thaJeztah) [inline daemon/prune.go]: Should the `!` actually be part of the filter name? Not sure how it's handled later on, but looks a bit tricky (i.e. `label!=foo.bar` would be accepted, but not `label != foo.bar`, correct?)

File: daemon/prune.go (hunk 1 above)
Comment (gdevillele) [inline daemon/prune.go]: Thanks. 
Might be a candidate for another PR.

File: daemon/prune.go (hunk 2 above)
Comment (mlaventure) [inline daemon/prune.go]: I don't think the later is a valid syntax for filters (e.g. `filters="label != foo.bar"`)

**Review by mlaventure — APPROVED** [review]
LGTM

File: daemon/prune.go (hunk 2 above)
Comment (gdevillele) [inline daemon/prune.go]: @thaJeztah 
- I agree that this is strange, following our discussion, I will open another PR to rewrite this logic.
- I don't think `filters` can contain any space, either.

Comment (thaJeztah) [inline daemon/prune.go]: Looked up the PR that implemented it, and indeed the format is `label!=key=val` or `label!=key`. Negative matches on _value_ alone don't seem to be supported (i.e., no `label=key!=val`), nor spaces in between (`label != key=val` etc.)

https://github.com/yongtang/docker/blob/702524732427ce028277f99f215e1fab297e6001/cli/command/utils.go#L95-L116

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
@gdevillele can you open an issue for tracking the changes on the way we match the filters? (perhaps write out which formats we _want_ to support,  if we want to support whitespace)

**Comment by gdevillele** [comment]
@thaJeztah will do. Thanks for merging!
